File: pregunta1/recursiva/fiboRecBase3.hs
fibo::Int->Int

fibo 1 = 0
fibo 2 = 1 
fibo 3 = 1
fibo n = fibo (n-1) + fibo(n-2) + fibo(n-3)

listFibonnaci::Int->[Int]
listFibonnaci n = [fibo i | i <- [1..n]]

File: pregunta1/recursiva/fiboRecBase3.scala
object Fibonacci {
    def fibo(n : Int) : Int={
        if (n == 1) return 0
        if (n == 2 || n == 3) return 1
        return fibo(n-1)+fibo(n-2)+fibo(n-3)
    }

    def list(n : Int) : String = {
        var cad = ""
        for (x <- 1 to n){
          cad += fibo(x) + " "
        }
        return cad
    }
}